packages/ui/react-ui-table/src/components/Cell/Cell.tsx

```diff
@@ -18,11 +18,45 @@ const CELL_NAME = 'Cell';
 const FocusableCell = <TData extends RowData, TValue>({ cell, children }: CellProps<TData, TValue>) => {
   const tableContext = useTableContext();
   const domAttributes = useFocusableGroup({ tabBehavior: 'limited' });
-
   const className = tdRoot(tableContext, cell.column.columnDef.meta?.cell?.classNames);
 
+  const handleKeyDown = useCallback((event: React.KeyboardEvent<HTMLTableCellElement>) => {
+    if (event.key !== 'Enter') {
+      return;
+    }
+
+    const currentCell = event.currentTarget;
+    const currentRow = currentCell.closest('tr');
+    if (!currentRow) {
+      return;
+    }
+
+    const cellIndex = Array.from(currentRow.cells || []).indexOf(currentCell);
+
+    const focusNextCell = (nextRow: HTMLTableRowElement) => {
+      const nextCell = nextRow.cells[cellIndex];
+      nextCell.focus();
+    };
+
+    const nextSibling = currentRow.nextElementSibling;
+    if (nextSibling instanceof HTMLTableRowElement) {
+      focusNextCell(nextSibling);
+    } else {
+      // Poll for the new 'add row' row to be added to the DOM.
+      const pollForNewRow = () => {
+        const newNextSibling = currentRow.nextElementSibling;
+        if (newNextSibling instanceof HTMLTableRowElement) {
+          focusNextCell(newNextSibling);
+        } else {
+          requestAnimationFrame(pollForNewRow);
+        }
+      };
+      requestAnimationFrame(pollForNewRow);
+    }
```

Comment:
Oof, a shame this is necessary 🥲

Reply:
And yet... we must.

Reply:
Actually I should probably add a max depth to this because it will break the editable storybooks.

Reply:
no way to set state so that when the new row is rendered it's autofocuesd?